src/Stage/Emit.elm

```diff
@@ -0,0 +1,437 @@
+module Stage.Emit exposing
```

Comment:
The Emit stage also got pretty refactored, to be more generic. The PrepareForBackend stage kicked the bucket, as we want to be generic and not model the whole compiler according to what makes sense in JS world.